src/app/common/NavTabs.vue

```diff
@@ -22,49 +22,48 @@
 </template>
 
 <script lang="ts" setup>
-import { KTabs, Tab } from '@kong/kongponents'
-import { PropType, computed } from 'vue'
-import { useRoute } from 'vue-router'
+import { computed } from 'vue'
+import { useRoute, type RouteLocationNormalizedLoaded } from 'vue-router'
 
-export interface NavTab {
-  title: string
+import type { StringNamedRouteRecordRaw } from '@/app/application/components/route-view/RouteView.vue'
+import { useI18n } from '@/utilities'
+import type { Tab } from '@kong/kongponents'
+
+interface NavTab extends Tab {
   routeName: string
-  module: string
+  isActive: boolean
 }
 
-const route = useRoute()
+const { t } = useI18n()
+const currentRoute = useRoute()
 
-const props = defineProps({
-  tabs: {
-    type: Array as PropType<NavTab[]>,
-    required: true,
-  },
+const props = withDefaults(defineProps<{
+  children: StringNamedRouteRecordRaw[]
+  active?: (route: RouteLocationNormalizedLoaded) => StringNamedRouteRecordRaw | undefined
+  i18nPrefix: string
+  filterPredicate?: (route: StringNamedRouteRecordRaw) => boolean
+}>(), {
+  active: () => undefined,
+  filterPredicate: () => true,
 })
 
-const kTabs = computed<Tab[]>(() => props.tabs.map((tab) => ({
-  title: tab.title,
-  hash: '#' + tab.routeName,
-})))
-const currentTabHash = computed(() => {
-  const modules = route.matched
-    .map((route) => route.meta.module ?? '')
-    .filter((module) => module !== '')
-  modules.reverse()
-
-  const activeTab = props.tabs.find((tab) => {
-    if (tab.routeName === route.name) {
-      return true
-    }
+const tabs = computed<NavTab[]>(() => {
+  return props.children
+    .filter(props.filterPredicate)
+    .map((route) => {
+      const routeName = route.name
+      const hash = '#' + routeName
+      const title = t(`${props.i18nPrefix}.${routeName}`)
+      const isActive = props.active(currentRoute)?.name === routeName
 
-    if (modules.includes(tab.module)) {
-      return true
-    }
+      return { title, hash, routeName, isActive }
+    })
+})
```

Comment:
If we pretend that we are using `DataCollection` to do filtering outside this component already (i.e. meaning the `filter` isn't required), is this computed function just reshaping data to be fed to `KTabs`, it kinda looks that way to me?

Looks like we could do?

```vue
<KTabs
  :tabs="items.map(item => ({
      title: t(`path.${item.name}`),
      hash: `#${item.name}`,
    }))"
  :model-value="`#${route.active(active)?.name ?? ''}`"
  hide-panels
  data-testid="nav-tabs"
>
  <template
    v-for="item in items"
    :key="`${item.name}`"
    #[`${item.name}-anchor`]
  >
    <RouterLink
      :data-testid="`${item.name}-tab`"
      :to="{ name: item.name }"
    >
      {{ t(`path.${item.name}`) }}
    </RouterLink>
  </template>
</KTabs>
```

Reply:
> What do you mean by "virtual code"?

I meant "inlining all virtually code"

> Why would we want to use `computed` here?

It cashes its results and unchanging dependencies won’t result in a recompuation and therefor prevent unnecessary rerenderings.

Reply:
Do you think we should cache the result of a `map` across a list of 5 or 6 static items?